Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -173,13 +186,17 @@ - (nullable id)valueForField:(id)field options:(FIRSnapshotOptions *)options {
   }
 
   FSTFieldValue *fieldValue = [[self.internalDocument data] valueForPath:fieldPath.internalValue];
-  return fieldValue == nil
-             ? nil
-             : [self convertedValue:fieldValue
-                            options:[FSTFieldValueOptions
-                                           optionsForSnapshotOptions:options
-                                        timestampsInSnapshotsEnabled:
-                                            self.firestore.settings.timestampsInSnapshotsEnabled]];
+  FSTFieldValueOptions *options = [self optionsForServerTimestampBehavior:serverTimestampBehavior];
+  return fieldValue == nil ? nil : [self convertedValue:fieldValue options:options];
+}
+
+- (FSTFieldValueOptions *)optionsForServerTimestampBehavior:
+    (FIRServerTimestampBehavior)serverTimestampBehavior {
+  FSTServerTimestampBehavior internalBehavior =
+      InternalServerTimestampBehavor(serverTimestampBehavior);
```

Comment:
Have you considered inlining this method call? It's only used once and would make it more apparent that `optionsForServerTimestampBehavior` is doing more than calling a constructor.

Reply:
I had pulled it out thinking there would be more than one caller. I don't think it really hurts anything to have it separate and it's exactly the kind of thing that could get reused so I don't see any huge reason to change it.

If anything I think if I'm going to expend any effort at all toward this I would just delete the internal version of the enum to remove the need for translation altogether.